tree-2.py

```diff
@@ -0,0 +1,81 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+
+#Problem 1:Construct Binary Tree from inorder and Poster order Traversal
+#Time Complexity O(n)
+#space complexity O(n)
```

Comment:
Space complexity is O(H) here in worst case. We build tree, but it is the result. So, we consider only recursive stack space here for space complexity